src/firefly/java/edu/caltech/ipac/firefly/server/dpanalyze/SofiaAnalyzer.java

```diff
@@ -78,21 +78,36 @@ public FileAnalysisReport analyzeFits(FileAnalysisReport inputReport,
                 e.printStackTrace();
             }
         }
-        if ( inst.equals("GREAT") && level.equals("LEVEL_4")){
-            try {
-                return convertToFrequencyParts(inputReport, inFile.getAbsolutePath());
-            } catch (FitsException e) {
-                e.printStackTrace();
-            } catch (IOException e) {
-                e.printStackTrace();
-            } catch (Exception e) {
-                e.printStackTrace();
+        if ( inst.equals("GREAT") ) {
+            if (level.equals("LEVEL_4")) {
+                try {
+                    return convertToFrequencyParts(inputReport, inFile.getAbsolutePath());
+                } catch (FitsException e) {
+                    e.printStackTrace();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                } catch (Exception e) {
+                    e.printStackTrace();
+                }
+            } else  if (level.equals("LEVEL_1")){
+
+              return setReportDisplay(inputReport, FileAnalysisReport.ChartTableDefOption.showChart);
             }
         }
 
         return inputReport;
     }
 
+    private FileAnalysisReport setReportDisplay(FileAnalysisReport report,FileAnalysisReport.ChartTableDefOption option ){
+
+        FileAnalysisReport retReport =  report.copy();
```

Comment:
I'm not sure that you even need to copy the report to set the parts with the default showing.

Reply:
I saw you used the copy and just kept the same manner.  I don't think that we need to make copy any places at all.  If you are OK, I will remove all the copy and update the input directly.  Again, it is your call.

Reply:
Ok, please,  if the copy is not needed, we should remove it, and test everything again before merging. Thank you.

Reply:
> Ok, please, if the copy is not needed, we should remove it, and test everything again before merging. Thank you.

I did quite some clean up.   Removed the unnecessary copy's and re-arranged the codes.  I tested and seemed all OK.